### DISCUSSION TITLE
Can't reset a component with eloquent model

### DISCUSSION THREAD

**Discussion by mokhosh** [livewire/livewire#3077] [7. Bugs (archived)]
Whenever I have an eloquent model as a public property on my component I can't call `$this->reset()` or I will get this:
```
Typed property App\Http\Livewire\Component::$model must not be accessed before initialization
```

**Comment by basepack** [comment]
@mokhosh While this is probably a valid issue, I always thought this was the Livewire way. I'd almost always have my own `init()` method where I do add this:

```php

$this->model = Product::make(); // or set it to whatever your use case is
$this->collection = collect(); // set collections too
```

**Comment by mokhosh** [comment]
That's what I have to do as well, but since @calebporzio is an angel who wants developers to write less and less code, and worry about less things, here's an area we can work on.

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by mokhosh** [comment]
REOPEN

**Comment by calebporzio** [comment]
Yep I'm confirming this is a bug.

PHP typed properties are really finicky and you have to add `isset()` checks before accessing them if you don't want to through those errors.

Because this is a relatively easily fixed issue and no one else has mentioned it. I'd welcome a PR for the fix @mokhosh if you're willing. But am going to close this.

**Comment by mokhosh** [comment]
@calebporzio I can think of two ways to do this.
1. initialize all typed attributes that are children of `Model` when we reset
2. return a new instance instead of null for typed properties in `InteractsWithProperties`@`getInitializedPropertyValue`

Sounds good?

**Comment by tomsgad** [comment]
Just got hit with the same error. Am not sure how to fix it. I have been trying and am creating more problems than solving them, Please help!

**Comment by tomsgad** [comment]
REOPEN

**Comment by mokhosh** [comment]
Make the type nullabale and initialize after you reset until we have a better solution
